server/chat-formatter.ts

```diff
@@ -444,3 +444,14 @@ class TextFormatter {
 export function formatText(str: string, isTrusted = false) {
 	return new TextFormatter(str, isTrusted).get();
 }
+
+/**
+ * Takes a string and strips all standard chat formatting from it, the text of a link is kept.
+ */
+export function stripFormatting(str: string) {
+	// Doesn't match > meme arrows because the angle bracket appears in the chat still.
+	str = str.replace(/\*\*([^\s\*]+)\*\*|__([^\s_]+)__|~~([^\s~]+)~~|``([^\s`]+)``|\^\^([^\s\^]+)\^\^|\\([^\s\\]+)\\/g,
```

Comment:
This is close to working, but still wont properly strip nested formatting such as `**~~text~~**`, which will return `~~text~~`. Any help on this?

Reply:
I guess checking if the RegExp still matches after the `.replace` and if it does running the function recursively should work.

Reply:
I think it's unnecessary to strip nested formatting. Links are the main thing, anyway.

Reply:
Like I would consider this "good enough" and merge despite that problem. No need to make it perfect.

Reply:
Ok, lets go with this.